toolbar/core/src/app.tsx

```diff
@@ -3,22 +3,17 @@ import './app.css';
 import { ContextProviders } from './components/context-providers';
 import { HotkeyListener } from './components/hotkey-listener';
 import { DesktopLayout } from './components/layouts/desktop';
-import { ClickBlocker } from './components/click-blocker';
 import { FocusLock } from './components/focus-lock';
 import { VisibilityManager } from './components/visibility-manager';
-import { useAppState } from './hooks/use-app-state';
+import { AppStateProvider } from './hooks/use-app-state';
 import type { ToolbarConfig } from './config';
+import { MainAppBlocker } from './components/main-app-blocker';
 
 export function App(config?: ToolbarConfig) {
-  const isMainAppBlocked = useAppState((state) => state.isMainAppBlocked);
-
   return (
-    <>
+    <AppStateProvider>
```

Comment:
@glenntws we use the `<AppStateProvider />` twice (it's also used in `context-providers.tsx`) - is this intended?